backend/drm/drm.c

```diff
@@ -593,17 +618,16 @@ static bool drm_connector_set_cursor(struct wlr_output *output,
 	struct wlr_drm_backend *drm = get_drm_backend_from_backend(output->backend);
 
 	struct wlr_drm_crtc *crtc = conn->crtc;
-	if (!crtc) {
-		return false;
+	if (crtc == NULL) {
+		goto allocate_fail;
 	}
 
 	struct wlr_drm_plane *plane = crtc->cursor;
 	if (!plane) {
 		// We don't have a real cursor plane, so we make a fake one
 		plane = calloc(1, sizeof(*plane));
 		if (!plane) {
-			wlr_log_errno(WLR_ERROR, "Allocation failed");
-			return false;
+			goto allocate_fail;
```

Comment:
Since `allocate_fail` has been declared it makes sense to me to use it. You say we lose line information in the debug output which is unfortunate, but it's a compromise.